packages/jest-haste-map/src/ModuleMap.ts

```diff
@@ -36,6 +35,30 @@ export default class ModuleMap {
   private readonly _raw: RawModuleMap;
   private json: SerializableModuleMap | undefined;
 
+  private static mapToArrayRecursive(
```

Comment:
You're probably wondering: wtf is this?

Turns out the `ModuleMap` wasn't being serialized in watch mode correctly but tests didn't catch it because we had 2 code paths.

1 code path = bug caught and fixed.